src/main/webapp/go.js

```diff
@@ -150,30 +185,39 @@ function toggleHintButton(hide) { //eslint-disable-line
  * Disable lint check because getHint() is called from go.html.
  */
 function getHint() { //eslint-disable-line
-  if (destIndex >= 0) {
-    const hintArr = huntArr[destIndex].hints;
+  if (hunt.getDestIndex() >= 0) {
+    const hintArr = hunt.getCurHints();
+    const hintIndex = hunt.getHintIndex();
     if (hintIndex < hintArr.length) {
       updateMessage(HINT_DISPLAY, hintArr[hintIndex]);
-      hintIndex++;
+      hunt.setHintIndex(hintIndex + 1);
     }
   }
 }
 
 /**
  * Determines whether the user entered the correct destination, and
  * adjusts the display accordingly.
+ * Disable lint check because handleDestinationAnswer() is called
+ * from go.html.
  */
-function handleDestinationAnswer() {
-  fetch(GUESS_URL).then((response) => response.json()).then((guess) => {
-    if (guess === huntArr[destIndex].name) {
-      toggleProceedButton(/* hide = */ false);
-      toggleHintButton(/* hide = */ true);
+function handleDestinationAnswer() { //eslint-disable-line
+  if (hunt.getDestIndex() == -1) { // User has not yet started the hunt.
+    return;
+  }
+  const userGuess = document.getElementById(GUESS_DISPLAY).value;
+  const queryStr = GUESS_URL + '?user-input=' + userGuess +
+      '&answer=' + hunt.getCurName();
+  fetch(queryStr).then((response) => response.json()).then((result) => {
+    if (result) {
+      toggleProceedButton(false);
+      toggleHintButton(true);
       updateMessage(SUBMIT_DISPLAY, CORRECT_MSSG);
-      updateMessage(RIDDLE_DISPLAY, huntArr[destIndex].name + ': ' +
-          huntArr[destIndex].description);
-      addMarkerToMap(huntArr[destIndex].lat, huntArr[destIndex].lng,
-          huntArr[destIndex].name);
-    } else if (guess.length != 0) {
+      updateMessage(RIDDLE_DISPLAY, hunt.getCurName() + ': ' +
+          hunt.getCurDescription());
+      addMarkerToMap(hunt.getCurLat(), hunt.getCurLng(),
+          hunt.getCurName());
+    } else if (userGuess.length != 0) {
```

Comment:
This only happens if they don't guess anything?  Shouldn't this just be an else?

Reply:
Yes! I initially added an else if to prevent the 'Wrong, try again!' method from displaying when the user doesn't type anything in, but it seems like handleDestinationAnswer() will only be called when the user hits the submit button.